handlers/backend_handler.go

```diff
@@ -140,15 +161,18 @@ func (bt *backendTransport) RoundTrip(req *http.Request) (resp *http.Response, e
 		if netErr, ok := err.(net.Error); ok {
 			if netErr.Timeout() {
 				logDetails["status"] = 504
+				responseCode = 504
```

Comment:
It's a shame we're not making use of the [constants](https://golang.org/pkg/net/http/#pkg-constants) provided in the http package here. Would be a nice change for another PR.

Reply:
I think this could be within the purview of this PR